pkg/cli/flags.go

```diff
@@ -1215,6 +1221,7 @@ func extraServerFlagInit(cmd *cobra.Command) error {
 		serverCfg.DisableTLSForHTTP = true
 	}
 	serverCfg.HTTPAddr = net.JoinHostPort(serverHTTPAddr, serverHTTPPort)
+	serverCfg.HTTPAdvertiseAddr = net.JoinHostPort(serverHTTPAdvertiseAddr, serverHTTPAdvertisePort)
```

Comment:
maybe before this line, for symmetry with the above:
```go
	if serverHTTPAdvertiseAddr == "" {
		if advSpecified {
			serverHTTPAdvertiseAddr = serverAdvertiseAddr
		} else {
			serverHTTPAdvertiseAddr = serverHTTPAddr
		}
	}
	if serverSQLAdvertisePort == "" {
		if advSpecified {
			serverHTTPAdvertisePort = serverAdvertisePort
		} else {
			serverHTTPAdvertisePort = serverHTTPPort
		}
	}
```

wdyt?

Reply:
SGTM. Done.

There's a bit of duplication of this in `addr_validation.go/ValidateAddrs`, or is that function doing something different?

Reply:
Last time I tried to unify the two parts together, I was getting strange failures in various unit tests + acceptance tests. I did not investigate further at the time.

Reply:
Hmm upon a second look I'm not sure about this.

The code above applies precedence in this order: CLI flag -> advertiseAddr -> httpAddr, but the `ValidateAddrs` function lets the httpAddr (listen address) take precedence over advertiseAddr. So this would create situations where if someone had both addresses set, the result would be different.

I added a more detailed test that also exercises the `ValidateAddrs` function which confirms this. I assume we don't want the same flags causing modified behavior after an upgrade.